lib/icinga/apiactions.cpp

```diff
@@ -458,11 +459,16 @@ Dictionary::Ptr ApiActions::ScheduleDowntime(const ConfigObject::Ptr& object,
 			if (allServices && !childService) {
 				ArrayData childServiceDowntimes;
 
-				for (const Service::Ptr& hostService : host->GetServices()) {
+				for (const Service::Ptr& childService : childHost->GetServices()) {
+					if (allChildren.find(static_pointer_cast<Checkable>(childService)) != allChildren.end()) {
+						/* The service is a child itself, no need to schedule a second downtime for it. */
+						continue;
+					}
+
 					Log(LogNotice, "ApiActions")
-						<< "Creating downtime for service " << hostService->GetName() << " on child host " << host->GetName();
+						<< "Creating downtime for service " << childService->GetName() << " on child host " << childHost->GetName();
 
-					Downtime::Ptr serviceDowntime = Downtime::AddDowntime(hostService, author, comment, startTime, endTime,
+					Downtime::Ptr serviceDowntime = Downtime::AddDowntime(childService, author, comment, startTime, endTime,
 						fixed, triggerName, duration);
```

Comment:
> Will cancelling the parent downtime now also cancel the service downtimes of children?

Looks like it won't as #8913 didn't add the information about that relation to this call. But I think it should, so I'll change that.

Reply:
Looking at the code, it's not only missing in that place but also in line 442. @N-o-X was this a deliberate decision to only set the parent relation for those downtimes created by `all_services` but not `child_options`?

Reply:
Thinking about it, I think I can remember the discussion we had about that (looks like we haven't written down the result over there in the PR).

We decided on not propagating the deletion down to dependencies as it's not clear what should happen if a downtime somewhere in the middle of a dependency chain is deleted. Intuitively, one could say that this should delete all the downtimes in that sub-tree, however dependencies are a graph, not a tree, so a dependant child could be in there due to multiple dependencies.

What's definitely an oversight in the current PR that service downtimes on child hosts don't have a relation to the downtime on the host. This should be addressed in any case. I'm no longer sure about dependencies. If we want to also delete then, this should be handled separately as this needs larger changes to what we store and how we handle deletes (as described in the second paragraph) and also a downtime can only one parent right now, so it can't have a relation to both the downtime on the child host and the root downtime.

Reply:
Aye fine, but then should the documentation get updated (https://icinga.com/docs/icinga-2/latest/doc/16-upgrading-icinga-2/#deletion-of-child-downtimes-on-services) before this fix is propagated. Otherwise the expected result is false.

Reply:
I can't see anything wrong with the documentation. It says `all_services` flag and nothing about `child_options`.

Reply:
But with this change the all_services flag applies also to childs. If that's the case, and I re-read the documentation with this in mind, I expect that these service downtimes of children also get removed.

Reply:
Deleting a downtime of a child host will now also delete downtimes for service of that particular child host.